ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/OptimizeService.java

```diff
@@ -397,15 +397,20 @@ private void initOptimizeTasksIntoOptimizeQueue() {
 
   @Override
   public void addNewTables(List<TableIdentifier> toAddTables) {
+    addNewTables(toAddTables, null);
+  }
+
+  private void addNewTables(List<TableIdentifier> toAddTables, ExecutorService executorService) {
```

Comment:
I think we donot need one more `addNewTables` and `clearRemovedTables` functions
```
@Override
  public void addNewTables(List<TableIdentifier> toAddTables) {
  if (!inited) {
    LOG.info("OptimizeService init not completed, can't add tables");
    return;
  }
  parallelProcessTables(toAddTables, table ->
          checkIfEnabledSelfOptimizing(com.netease.arctic.ams.server.utils.CatalogUtil
              .getArcticCatalog(table.getCatalog())
              .loadTable(table)), null);
  LOG.info("add tables[{}] {}", toAddTables.size(), toAddTables);
  }
```